### ISSUE TITLE
"Uncaught (in promise)" error in console in incognito

### ISSUE THREAD

**Issue by whispy** [keepassxreboot/keepassxc-browser#62]
When using KeepassXC browser in Chrome (v64) and browsing in incognito mode, the extension throws an error in the console.

## Expected Behavior
I would expect there to be no console error.

## Current Behavior
See above.

## Steps to Reproduce (for bugs)
1. Be using KeepassXC Browser in Chrome with the extension set to `Allow in incognito`.
2. Browse to a site in an incognito window (www.reddit.com seems to reliably cause the error).
3. Open console and view error.

## Debug info
```
Uncaught (in promise) {message: "The message port closed before a response was received."}

  | n | @ | browser-polyfill.min.js:1
  | sendResponseAndClearCallback | @ | VM30 extensions::messaging:417
  | disconnectListener | @ | VM30 extensions::messaging:441
  | EventImpl.dispatchToListener | @ | VM23 extensions::event_bindings:403
  | publicClassPrototype.(anonymous function) | @ | VM29 extensions::utils:138
  | EventImpl.dispatch_ | @ | VM23 extensions::event_bindings:387
  | EventImpl.dispatch | @ | VM23 extensions::event_bindings:409
  | publicClassPrototype.(anonymous function) | @ | VM29 extensions::utils:138
  | dispatchOnDisconnect | @ | VM30 extensions::messaging:378
```

## General Info
KeePassXC fork - 2.3.0
keepassxc-browser - 1.0.1

Operating system: Mac
Browser: Chrome (v64)
Proxy used: unknown

**Comment by varjolintu** [comment]
It's a known issue. However, it doesn't crash the script or cause anything to actually fail, or doesn't it?

**Comment by whispy** [comment]
I don't notice any issues with the extension's functionality when the error occurs. I didn't see any open issues relating to it.

I spend most of my work time with the console open for development, so any errors catch my eye.

**Comment by kronenpj** [comment]
I'm seeing this error on a normal tab (not incognito) on Windows. Is this similar enough to report here or should another issue be opened?

**Comment by varjolintu** [comment]
These kinds of "errors" are quite common. Sometimes the receiving end has been already deleted (tab etc).

**Comment by jcrben** [comment]
I saw this today while not in incognito and it was correlated with a neverending spinner over at https://www.qplum.co/qfolio/progress/0

But when I disabled and re-enabled, I couldn't replicate the issue.

I should note that another error that I see commonly (`Uncaught (in promise) {message: "The message port closed before a response was received."}`) did NOT cause the spinner

![image](https://user-images.githubusercontent.com/5614134/38773735-f680b7d0-4008-11e8-89f9-13f89fb4e2ee.png)

**Comment by varjolintu** [comment]
@jcrben As you can see these errors are generated by Mozilla's own browser-polyfill.

**Comment by varjolintu** [comment]
@jcrben See [this page](https://github.com/mozilla/webextension-polyfill).

> This library allows extensions written for the Promise-based WebExtension/BrowserExt API being standardized by the W3 Browser Extensions group to be used without modification in Google Chrome.

This is why you see it with Chromium-based browsers.

**Comment by jcrben** [comment]
Ah. Looks like this was originally filed in https://github.com/mozilla/webextension-polyfill/issues/58 which is referenced in https://github.com/mozilla/webextension-polyfill/issues/65

**Comment by varjolintu** [comment]
I'm going to test updating the plugin. Now we use a custom version with a few hotfixes.

**Comment by hrobertson** [comment]
Since 1.1.0 the console is getting flooded with this on several sites I visit.
For example I load soundcloud.com (already logged in) and with no further interaction I get about 20 instances of `{message: "The message port closed before a response was received."}` in the console from `keepassxc-browser.js:1264`.
A minute later I get another 30 of the above, plus 8 of `Uncaught (in promise) {message: "The message port closed before a response was received."}` from `browser-polyfill.min.js`.

Prior to 1.1.0 I did not have this issue.

**Comment by varjolintu** [comment]
The latest webextension-polyfill solves this issue but this issue must be fixed before we can update it to the extension: https://github.com/mozilla/webextension-polyfill/pull/153.
